sdks/java/harness/src/main/java/org/apache/beam/fn/harness/control/ProcessBundleHandler.java

```diff
@@ -255,6 +258,19 @@ private void createRunnerAndConsumersForPTransformRecursively(
       // Create a BeamFnStateClient
       for (Map.Entry<String, RunnerApi.PTransform> entry :
           bundleDescriptor.getTransformsMap().entrySet()) {
+
+        if (RemoteGrpcPortWrite.isSinkPtransform(entry.getValue())) {
```

Comment:
I thought you cared that all RemoteGrpcPortReads had the same ApiServiceDescriptor, not the writes.

Reply:
I think you are right, but probably they were the same, which is why this worked so far.